MobileWallet/Screens/Settings/BackUpSettings/BackupWalletSettingsViewController.swift

```diff
@@ -39,103 +39,177 @@
             switch self {
-            case .backUpToiCloud: return NSLocalizedString("backup_wallet_settings.item.to_icloud", comment: "BackupWalletSettings view")
+            case .iCloudBackups: return NSLocalizedString("backup_wallet_settings.item.icloud_backups", comment: "BackupWalletSettings view")
+            case .setupPassword:
+                if Migrations.loadBackupPasswordFromKeychain() != nil {
+                    return NSLocalizedString("backup_wallet_settings.item.change_password", comment: "BackupWalletSettings view")
+                } else {
+                    return NSLocalizedString("backup_wallet_settings.item.secure_your_backup", comment: "BackupWalletSettings view")
+                }
+            case .backUpNow: return NSLocalizedString("backup_wallet_settings.item.backup_now", comment: "BackupWalletSettings view")
             case .backUpWithRecoveryPhrase: return NSLocalizedString("backup_wallet_settings.item.with_recovery_phrase", comment: "BackupWalletSettings view")
             }
         }
     }
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        backUpWalletItem = backupNowSectionItems.first(where: { $0.title == BackupWalletSettingsItem.backUpNow.rawValue })
+        iCloudBackupsItem = settingsSectionItems.first(where: { $0.title == BackupWalletSettingsItem.iCloudBackups.rawValue })
+
         tableView.delegate = self
         tableView.dataSource = self
-
-        NotificationCenter.default.addObserver(self, selector: #selector(updateMarks), name: UIApplication.willEnterForegroundNotification, object: nil)
+        observeICloudBackupsSwitcher()
     }
 
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        updateMarks()
+    private func onBackupNowAction() {
+        createWalletBackup()
     }
 
-    private func onBackupToiCloudAction() {
-        let localAuth = LAContext()
-        localAuth.authenticateUser(reason: .userVerification) { [weak self] in
-            do {
-                try self?.iCloudBackup.createWalletBackup()
-                self?.iCloudBackupItem.mark = .progress
-            } catch {
-                UserFeedback.shared.error(title: NSLocalizedString("iCloud_backup.error.title", comment: "iCloudBackup error"), description: "", error: error)
-                self?.iCloudBackupItem.mark = .attention
-            }
-        }
+    private func onChangePasswordAction() {
+        navigationController?.pushViewController(SecureBackupViewController(), animated: true)
     }
 
     private func onBackupWithRecoveryPhraseAction() {
         navigationController?.pushViewController(SeedPhraseViewController(), animated: true)
     }
 
-    @objc private func updateMarks() {
-        if iCloudBackup.inProgress {
-            iCloudBackupItem.mark = .progress
-            iCloudBackupItem.percent = iCloudBackup.progressValue
-            return
+    private func observeICloudBackupsSwitcher() {
```

Comment:
I think you can call it just `Switch` instead of `Switcher` everywhere.

Reply:
I can't. 'switch' is key word in swift language.

Reply:
Ok I thought you're not using the word on its own but always with other words such as `backupSwitch`.

Reply:
`switch` on its own is reserved but wouldn't `observeICloudBackupsSwitch` be ok?